File: android/src/main/java/com/reactnativecouchbaselite/RNDatabase.kt
package com.reactnativecouchbaselite

import com.couchbase.lite.Database
import com.couchbase.lite.SafeReadableMap

object RNDatabase {
  private val databaseCache: MutableMap<String, Database> = hashMapOf()

  fun get(name: String): Database =
    databaseCache.getOrElse(name, {
      val db = Database(name)
      databaseCache[name] = db
      db
    })

  fun decode(obj: SafeReadableMap) = obj.getString("name")
    .map { get(it) }
}

File: ios/RNDatabase.swift
import Bow
import CouchbaseLiteSwift

class RNDatabase {
  static var databaseCache: [String: Database] = [:]
  
  static func get(_ name: String) -> Database {
    databaseCache[name].toOption().getOrElse({
      let db = try! Database(name: name)
      databaseCache[name] = db
      return db
    })
  }
}
	
